commit 807a045ab7f51993a2c1b3116016cbbd4f3d20d6	807a045ab7f51993a2c1b3116016cbbd4f3d20d6
Author: Michael Niedermayer <michaelni@gmx.at>
Date:   Wed Jan 25 23:23:35 2012 +0100

    kgv1dec: Increase offsets array size so it is large enough.
    
    Fixes CVE-2011-3945
    
    Found-by: Mateusz "j00ru" Jurczyk and Gynvael Coldwind
    Signed-off-by: Michael Niedermayer <michaelni@gmx.at>

diff --git a/libavcodec/kgv1dec.c b/libavcodec/kgv1dec.c
index 398b8af0d5..4566e35b74 100644
--- a/libavcodec/kgv1dec.c
+++ b/libavcodec/kgv1dec.c
@@ -39,7 +39,7 @@ static int decode_frame(AVCodecContext *avctx, void *data, int *data_size, AVPac
     const uint8_t *buf = avpkt->data;
     const uint8_t *buf_end = buf + avpkt->size;
     KgvContext * const c = avctx->priv_data;
-    int offsets[7];
+    int offsets[8];
     uint16_t *out, *prev;
     int outcnt = 0, maxcnt;
     int w, h, i;
@@ -69,7 +69,7 @@ static int decode_frame(AVCodecContext *avctx, void *data, int *data_size, AVPac
         return -1;
     c->prev = prev;
 
-    for (i = 0; i < 7; i++)
+    for (i = 0; i < 8; i++)
         offsets[i] = -1;
 
     while (outcnt < maxcnt && buf_end - 2 > buf) {
